client/src/context/user/UserState.js

```diff
@@ -57,10 +64,39 @@ const UserState = (props) => {
     }
   }
 
+  // Search users
+  const searchUser = async (searchParam) => {
+    try {
+      if (searchParam === '' || searchParam === ' ') {
+        clearSearch()
```

Comment:
consider trimming the spacesthe input so that you can do a single comparator. https://www.w3schools.com/jsref/jsref_trim_string.asp